src/Text/Trifecta/Parser.hs

```diff
@@ -129,14 +130,23 @@ instance Semigroup a => Semigroup (Parser a) where
   (<>) = liftA2 (<>)
   {-# INLINE (<>) #-}
 
+{- In order to silence '-Wnoncanonical-monoid-instances' we'd need to
+   change the type-signature to emulate the not-yet existing superclass
+   relationship:
+
+instance (Semigroup a, Monoid a) => Monoid (Parser a) where
+  mappend = (<>)
+  mempty = pure mempty
+-}
 instance Monoid a => Monoid (Parser a) where
```

Comment:
@RyanGlScott I'm fine with breaking changes; I was just looking at `free`, and there it would require quite a bit of code-duplication if we don't make a breaking change (regarding class constraints).